src/coreclr/jit/lclvars.cpp

```diff
@@ -2933,6 +2938,8 @@ void Compiler::lvaSetStruct(unsigned varNum, ClassLayout* layout, bool unsafeVal
         }
 #endif // not TARGET_64BIT
 
+        varDsc->SetIsSpan(this->isSpanClass(layout->GetClassHandle()));
```

Comment:
Can you move this under an `opts.OptimizationEnabled()` check?

Reply:
Done

Reply:
It looks like we have an ordering issue -- we don't yet know if we will be optimizing or not when `lvaSetStruct` is called for some locals. Can you revert the change and I'll think about if we can move this somewhere else separately.

Reply:
Sure. I will do it tomorrow.